build/xtask/src/dist.rs

```diff
@@ -203,7 +228,11 @@ pub fn package(
         .keys()
         .map(|name| {
             let ep = if tasks_to_build.contains(name.as_str()) {
-                build_task(&cfg, name, &allocs, &mut all_output_sections)
+                // Link tasks regardless of whether they have changed, because
```

Comment:
Note that this technique is probably not sufficient for handling changes to the `rlink` script.